Show wordpress icons

jjrohrer opened the pull request:
I wanted more help in tracking where my plugins were coming from.  I added a chunk that showed the wordpress icon if the plugin would be updated from wordpress.org.  So, now plugins get an icon if they are updated via one of the three major repositories, but no icon if they are hosted elsewhere.  I think this is a nice way to track/segregate the 'quasi-public' plugins from the 'quasi-private' plugins. 

BTW - I'm fairly new to the git community - so I'll apologize in advance for everything stupid that I've done, like, ever.

![screenshot_101](https://cloud.githubusercontent.com/assets/69016/7006130/af67d3b0-dc4c-11e4-973a-280929d4b6ac.png)

jjrohrer committed:
pre download

jjrohrer committed:
Merge branch 'master' of https://github.com/jjrohrer/github-link

# By Viktor Szépe (3) and others
# Via Viktor Szépe (2) and carstenbach (1)
* 'master' of https://github.com/jjrohrer/github-link:
  Added german translation
  Added multisite support
  updated readme links
  Hungarian translation
  Add portuguese translation
  Add .pot catalog and description keys text file
  Add possibility to translation
  initial

jjrohrer committed:
Added wordpress icons of unkown origin.  Presumably ok to use, but not really sure.

jjrohrer committed:
Got wordpress.org icons to show at the begining to help clarify which plugins are from wp.org, which are from github/bitbucket, and which are custom or commercial.

jjrohrer committed:
Perhaps works for real, now, showing plugins that are updated via wordpress.org.

jjrohrer committed:
updated readme to reflect wordpress presence

jjrohrer committed:
update readme, again, to reflect presence of wordpress icons

jjrohrer committed:
temporarily point to jjrohrer since jj needs to update from here.  Hmm, should probably be a branch

szepeviktor commented:
OK. Very nice.
I am used to sending PR-s but not receiving.
Please try to develop locally and keep 1 commit per PR.

szepeviktor commented:
You could google for merging commits...

szepeviktor commented on github-link.php:
```diff
@@ -9,7 +9,8 @@
 Domain Path:       /languages
 Text Domain:       github-link
-GitHub Plugin URI: https://github.com/szepeviktor/github-link
+GitHub Plugin URI: https://github.com/jjrohrer/github-link
```
Please revert this.

jjrohrer replied:
Yike - my newness shows.  I thought a pull request would point to specific commit.  This was added later.  Reverting now.

szepeviktor commented on README.md:
```diff
@@ -17,6 +17,12 @@ the GitHub icon is displayed **after** other plugin actions (links), otherwise i
+
+### Wordpress.org headers
+
+- `Plugin URI`, if it contains the string 'wordpress.org', is shown as a Wordpress logo ![Wordpress logo](icon/wordpress_org_32x32.jpg)
```
This icon shold stay for linking it from the README.

szepeviktor replied:
Sorry. Use https://s.w.org/about/images/logos/wordpress-logo-32-blue.png
from https://wordpress.org/about/logos/

jjrohrer replied:
I think this is OBE by the dashicon

szepeviktor replied:
Needed for the readme.

szepeviktor commented on github-link.php:
```diff
@@ -9,7 +9,8 @@
 Text Domain:       github-link
-GitHub Plugin URI: https://github.com/szepeviktor/github-link
+GitHub Plugin URI: https://github.com/jjrohrer/github-link
+Depends: github-updater
```
Please remove this line.
I don't use GHU. Don't tell anyone!!
Maybe others aren't using GHU...

afragen replied:
I won't tell anyone. :wink:

szepeviktor commented on github-link.php:
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
 
+
+    // if from wordpress.org, show that now.
+    if ( ! empty( $plugin_data['url']) && strstr($plugin_data['url'],'wordpress.org') !== FALSE) {
```
WP coding style please:

``` php
false !== strstr( $plugin_data['url'], '//wordpress.org/plugins/' ) )
```

and split the two conditions into two lines as [above](https://github.com/szepeviktor/github-link/pull/3/files#diff-e570aa2cdf28400231de7f21099894b4L55).

jjrohrer replied:
done

szepeviktor commented on github-link.php:
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
+    // if from wordpress.org, show that now.
+    if ( ! empty( $plugin_data['url']) && strstr($plugin_data['url'],'wordpress.org') !== FALSE) {
+        $icon = "icon/wordpress_org_32x32.jpg";
+        $new_action = array ('wordpress_org' => sprintf(
```
Use a variable: `$wp_link_template` and copy $link_template's content. (title, link, target ...)

jjrohrer replied:
Sorry - I'm not following what you are suggesting.

szepeviktor replied:
Define a template like in `$link_template`.

szepeviktor commented on github-link.php:
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
+    if ( ! empty( $plugin_data['url']) && strstr($plugin_data['url'],'wordpress.org') !== FALSE) {
+        $icon = "icon/wordpress_org_32x32.jpg";
+        $new_action = array ('wordpress_org' => sprintf(
+            '<img src="%s" style="vertical-align:-3px" height="16" width="16" alt="%s" />',
```
Dashicons please. With color as inline style.

``` html
<span class="dashicons dashicons-wordpress"></span>
```

szepeviktor commented on github-link.php:
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
+        $new_action = array ('wordpress_org' => sprintf(
+            '<img src="%s" style="vertical-align:-3px" height="16" width="16" alt="%s" />',
+            plugins_url( $icon, __FILE__ ),
+            "A wordpress.org plugin."
```
`__( "Visit WordPress.org Plugin Page" , "github-link" )`
And generate pot, po and mo.

jjrohrer replied:
I've never done that - I wouldn't even know where to start, but I replaced the previous text with __(...)

szepeviktor commented on github-link.php:
```diff
@@ -105,5 +108,18 @@ function GHL_plugin_link( $actions, $plugin_file, $plugin_data, $context ) {
+            plugins_url( $icon, __FILE__ ),
+            "A wordpress.org plugin."
+        ) );
+        //$actions = $new_action + $actions;//ensure at front
```
Only `// ensure WP icon at front`.

jjrohrer committed:
Removed local wordpress icon because replaced with Dashicons
Reverted Plugin URI - oops
Update if to WP coding style
Added _Blank
Added __("Visit...

Not done - make .pot, po, & mo

jjrohrer committed:
Re-added wordpress icon from https://s.w.org/about/images/logos/wordpress-logo-32-blue.png for read

jjrohrer committed:
Moved inline link template for wp.org to a separate var.

szepeviktor commented:
There is a fundamental problem. PLugin authors put their own blog's URL in `Plugin URI`.
@afragen Could you help us to detect plugin origin **without an HTTP request**?

szepeviktor commented:
@afragen Just tell me I already know how to: https://github.com/szepeviktor/github-link/blob/master/github-link.php#L55-L58

szepeviktor commented on github-link.php:
```diff
@@ -10,6 +10,7 @@
 Domain Path:       /languages
 Text Domain:       github-link
 GitHub Plugin URI: https://github.com/szepeviktor/github-link
+Depends: GitHub Updater
```
I'll finish this PR. Thank you!!

afragen commented:
If you're trying to detect if the plugin is NOT using GHU, it's probably easier to check the `empty()` of `GitHub Plugin URI` and `Bitbucket Plugin URI` and then assume hosting on wp.org.

Also, `GitHub Access Token` header no longer exists.

jjrohrer commented:
I would specifically recommend against assuming wp.org http://wp.org/.  My own install, for example, is about 1/3 ‘other’.  They are privately hosted because they are commercial plugins or custom plugins still underdevelopment.  This is my original use-case for the WP icons - to find out which plugins where part of a central repository, vs which ones part of some other quasi-private repository.
JJ

> On Apr 6, 2015, at 1:23 PM, Andy Fragen notifications@github.com wrote:
> 
> If you're trying to detect if the plugin is NOT using GHU, it's probably easier to check the empty() of GitHub Plugin URI and Bitbucket Plugin URI and then assume hosting on wp.org.
> 
> Also, GitHub Access Token header no longer exists.
> 
> To ensure GHU is active you will probably need use Fragen\GitHub_Updater in this plugin and a class_exists check.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/szepeviktor/github-link/pull/3#issuecomment-90161953.

szepeviktor commented:
Thank you!

szepeviktor commented:
As I mentioned WP.org detection was already in the code. https://github.com/szepeviktor/github-link/blob/master/github-link.php#L56-L60

afragen commented:
That won't work. GHU adds to `$plugin_state->response[ $plugin_file ]` when an update is detected.

szepeviktor commented:
Oh no! What do you suggest? (what is not a HTTP request)

afragen commented:
If you take look at our last GHU commit re:View details. It seems that only .org plugins and now GHU plugins will show View details. Others will show 'Visit plugin site'. Some combination of checking his and having GitHub/Bitbucket Plugin URI empty could be a solution.

szepeviktor commented:
How about searching for `//wordpress.org/plugins/` in `$plugin_state->response[ $plugin_file ]->url`?

szepeviktor commented:
Noone should write:

```
GitHub Plugin URI: https://wordpress.org/plugins/post-tags-and-categories-for-pages/
```

or `https://my-site.net//wordpress.org/plugins/bingo`

szepeviktor commented:
``` php
// GHU also sets plugin->url
if ( false !== strstr( $plugin_uri, '//wordpress.org/plugins/' ) ) {
    $new_action = array( 'wordpress_org' => sprintf(
        $wp_link_template,
        $plugin_uri,
        __( "Visit WordPress.org Plugin Page" , "github-link" )
    ) );
    $actions = array_merge( $new_action, $actions );
}
```

afragen commented:
> How about searching for //downloads.wordpress.org/plugin/ in $plugin_state->response[ $plugin_file ]->url?

That should work. Don't forget to check $plugin_state->no_updates too.

szepeviktor commented:
Thank you!
Done. https://github.com/szepeviktor/github-link/commit/5a8c226224854887f4b8161b971b479ce7af3a9a#diff-e570aa2cdf28400231de7f21099894b4R117
